doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+    \\ \\
+    If the term is unused, then we will connect an eraser node to the second Auxiliary port.
+    \\ \\
+    This thus preserves the number of free ports.
+    \\ \\
+    By induction the subterms on Auxiliary one are also well formed
+    \\ \\
+    $\therefore$ the Lambda term is $α$ equivalent to the normal evaluation answer.
+  \end{tcolorbox}
+
+  \begin{tcolorbox}[title= Case 3 - FanIn, colframe=inner-box-1]
+    FanIn's are another interesting case, for this case we allocate a map from each FanIn to the traversal status.
+    \\ \\
+    For this, we keep a list of the following types
+    \begin{verbatim}
+     type Status = In FanPort
+                 | Complete FanPort
+     type FanPort = Circle | Star
+    \end{verbatim}
+    Where circle is the first auxiliary port and star is the second.
+
+    \begin{figure}[H]
+        \begin{tikzpicture}
+          \inetcell[]{a}{$f_i$}[5]
+          \draw (-0.29, 0.23) node {$\scalebox{1.9}{$\circ$}$};
+          \draw ( 0.29, 0.29) node {$\scalebox{1.7}{$\star$}$};
+          \cutWire{a.out}{b.north}{}{}
+        \end{tikzpicture}
+    \end{figure}
+
+
+    For this analysis we must consider what port we can enter from, however first we should note that before entering the primary port of a FanIn, that we must have first traversed through either $\star$ or $\circ$ first.
+    \\ \\
+    This is the case because entering a FanIn in this way is known as a FanOut, and denotes the end of sharing.
+    And since ending sharing before starting is not possible from a valid EAL-Net, these cases never happen.
+    \\ \\
+    \begin{tcolorbox}[title= Enter from Prim, colframe=inner-box-2]
+      As shown above, we must have already traversed a FanIn node, namely $\star$ or $\circ$.
+      \\ \\
+      This means the map from the FanIn number $i$ is not empty, and thus we have a history of completing nodes or being in nodes.
+      \\ \\
+      If we have an unfinished $In$, then we pick the most recent $In$ port left to traverse.
+      This is forced as this node is considered a FanOut and thus closes sharing.
+      \\ \\
+      We then mark the node as complete. If we have completed a single node, then we simply choose to leave through the other port.
+      \\ \\
+      Since we are working over valid EAL-Net's which require no oracle, there is no ambiguity in this mechanism. Furthermore, this fact also excludes any other configurations from happening.
+      \\ \\
+    \end{tcolorbox}
+    \begin{tcolorbox}[title= Enter from Aux, colframe=inner-box-2]
+      We entered the FanIn through an auxiliary port, so we mark in the second map that we are $In$ this port at FanIn node $i$.
+      \\ \\
+      We then leave through the principle port, denoting the beginning of sharing.
+    \end{tcolorbox}
+    These cases handle all possible configurations in which we can enter and the validity of the expressions are handled by induction.
```

Comment:
Which expressions?

Reply:
the read-back expression of running the algorithm

Reply:
I see, we should refer to induction separately in each case where we invoke it for a subterm I think.